pkg/providers/cloudstack/spec.go

```diff
@@ -11,6 +11,23 @@ type spec struct {
 	machineConfigsLookup map[string]*anywherev1.CloudStackMachineConfig
 }
 
+func NewSpec(clusterSpec *cluster.Spec, machineConfigs map[string]*anywherev1.CloudStackMachineConfig, datacenterConfig *anywherev1.CloudStackDatacenterConfig) *spec {
```

Comment:
Exported functions should return exported types. Any reason `spec` is unexported?